include/zephyr/sys/util.h

```diff
@@ -783,6 +783,34 @@ static inline void mem_xor_128(uint8_t dst[16], const uint8_t src1[16], const ui
 	mem_xor_n(dst, src1, src2, 16);
 }
 
+/**
+ * @brief Returns the number of bits set in a value
+ *
+ * @param value The value to count number of bits set of
+ * @param len The number of octets in @p value
+ */
+static inline size_t count_bits(const void *value, size_t len)
+{
+	const uint8_t *value_u8 = (const uint8_t *)value;
+	size_t cnt = 0U;
+
+	for (size_t i = 0U; i < len; i++) {
+		if (value_u8[i] != 0U) {
+#ifdef POPCOUNT
+			cnt += POPCOUNT(value_u8[i]);
+#else
+			/* Implements Brian Kernighan’s Algorithm to count bits */
```

Comment:
Is this comments still valid? Doesn't look like Brian Kernighan?

Reply:
There are few different implementations of it. This is the one the exists several places in Zephyr and also matches e.g. https://www.educative.io/answers/what-is-kernighans-algorithm 

The `while` loop is effectively the algorithm modified to work on only `uint8_t` values which is then done `n` times, `n` being the number of octets in the value, as `count_bits` supports any size value (including values larger than `uint64_t`).